src/compositemark/errorbar.ts

```diff
@@ -1,42 +1,395 @@
+        }
+      }
+    );
   }
 
   return {
     ...outerSpec,
+    transform,
     layer: [
+      ...errorBarSinglePointPartMarksSpec('bar'),
+      ...errorBarSinglePointPartMarksSpec('line'),
+      ...errorBarSinglePointPartMarksSpec('ticks', 'lower_rule'),
+      ...errorBarSinglePointPartMarksSpec('ticks', 'upper_rule'),
+      ...errorBarSinglePointPartMarksSpec('rule', 'lower_rule', 'upper_rule'),
+      ...errorBarSinglePointPartMarksSpec('point')
+    ]
+  };
+}
+
+function errorBarOrient(spec: GenericUnitSpec<Encoding<Field>, ErrorBar | ErrorBarDef>): Orient {
+  const {mark: mark, encoding: encoding, projection: _p, ..._outerSpec} = spec;
+
+  if (isFieldDef(encoding.x) && isContinuous(encoding.x)) {
+    // x is continuous
+    if (isFieldDef(encoding.y) && isContinuous(encoding.y)) {
+      // both x and y are continuous
+      if (encoding.x.aggregate === undefined && encoding.y.aggregate === ERRORBAR) {
+        return 'vertical';
+      } else if (encoding.y.aggregate === undefined && encoding.x.aggregate === ERRORBAR) {
+        return 'horizontal';
+      } else if (encoding.x.aggregate === ERRORBAR && encoding.y.aggregate === ERRORBAR) {
+        throw new Error('Both x and y cannot have aggregate');
+      } else {
+        if (isMarkDef(mark) && mark.orient) {
+          return mark.orient;
+        }
+
+        // default orientation = vertical
+        return 'vertical';
+      }
+    }
+
+    // x is continuous but y is not
+    return 'horizontal';
+  } else if (isFieldDef(encoding.y) && isContinuous(encoding.y)) {
+    // y is continuous but x is not
+    return 'vertical';
+  } else {
+    // Neither x nor y is continuous.
+    throw new Error('Need a valid continuous axis for errorbars');
+  }
+}
+
+
+function errorBarContinousAxis(spec: GenericUnitSpec<Encoding<string>, ErrorBar | ErrorBarDef>, orient: Orient) {
+  const {mark: mark, encoding: encoding, projection: _p, ..._outerSpec} = spec;
+
+  let continuousAxisChannelDef: PositionFieldDef<string>;
+  let continuousAxis: 'x' | 'y';
+
+  if (orient === 'vertical') {
+    continuousAxis = 'y';
+    continuousAxisChannelDef = encoding.y as FieldDef<string>; // Safe to cast because if y is not continuous fielddef, the orient would not be vertical.
+  } else {
+    continuousAxis = 'x';
+    continuousAxisChannelDef = encoding.x as FieldDef<string>; // Safe to cast because if x is not continuous fielddef, the orient would not be horizontal.
+  }
+
+  if (continuousAxisChannelDef && continuousAxisChannelDef.aggregate) {
+    const {aggregate, ...continuousAxisWithoutAggregate} = continuousAxisChannelDef;
+    if (aggregate !== ERRORBAR) {
+      log.warn(`Continuous axis should not have customized aggregation function ${aggregate}`);
+    }
+    continuousAxisChannelDef = continuousAxisWithoutAggregate;
+  }
+
+  return {
+    continuousAxisChannelDef,
+    continuousAxis
+  };
+}
+
+function isExtentCI(extent: ErrorBarExtent): extent is 'ci' {
```

Comment:
You shouldn't have to declare type guards for string literals.  Please remove both of this.